registry/service.go

```diff
@@ -115,6 +148,9 @@ func splitReposSearchTerm(reposName string) (string, string) {
 // Search queries the public registry for images matching the specified
 // search terms, and returns the results.
 func (s *DefaultService) Search(ctx context.Context, term string, limit int, authConfig *types.AuthConfig, userAgent string, headers map[string][]string) (*registrytypes.SearchResults, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
I know I suggested this, but now I'm not so sure. It means only one search can run at once, for example. And a running search will block calls to other `DefaultService` methods. I didn't realize that some of these methods are long-running.

I guess the general strategy of locking `s.config` in every exported method is okay as long as we release the lock before any long-running operation. Not sure if this just an issue for `Search`, or if it also applies to other methods.